src/org/labkey/test/tests/NonStandardDateAndTimeFormatTest.java

```diff
@@ -0,0 +1,1334 @@
+    private void validateSettingsPage(@Nullable Boolean dateInherited, String dateFormat, boolean dateWarning,
+                                      @Nullable Boolean timeInherited, String timeFormat, boolean timeWarning,
+                                      @Nullable Boolean dateTimeInherited, String dtDateFormat, String dtTimeFormat,
+                                      boolean dtDateWarning, boolean dtTimeWarning)
+    {
+
+        log("Check Date field settings.");
+
+        checker().verifyEquals("Format of Default Date is not as expected.",
+                dateFormat, getFormatFromControl("defaultDateFormat"));
+
+        if (null != dateInherited)
+        {
+            checker().verifyEquals("Inherited value for Date format not as expected.",
+                    dateInherited, getFieldInherited("defaultDateFormatInherited"));
+        }
+
+        checker().verifyEquals("Non-standard warning for Date field not as expected.",
+                dateWarning, getNonStandardWarning("defaultDateFormat"));
+
+
+        log("Check Time field settings.");
+
+        checker().verifyEquals("Format of Default Time is not as expected.",
+                timeFormat, getFormatFromControl("defaultTimeFormat"));
+
+        if (null != timeInherited)
+        {
+            checker().verifyEquals("Inherited value for Time format not as expected.",
+                    timeInherited, getFieldInherited("defaultTimeFormatInherited"));
+        }
+
+        checker().verifyEquals("Non-standard warning for Time field not as expected.",
+                timeWarning, getNonStandardWarning("defaultTimeFormat"));
+
+
+        log("Check DateTime field settings.");
+
+        checker().verifyEquals("Format of Default DateTime (Date) is not as expected.",
+                dtDateFormat, getFormatFromControl("dateSelect"));
+
+        checker().verifyEquals("Format of Default DateTime (Time) is not as expected.",
+                dtTimeFormat, getFormatFromControl("timeSelect"));
+
+        if (null != dateTimeInherited)
+        {
+            checker().verifyEquals("Inherited value for DateTime format not as expected.",
+                    dateTimeInherited, getFieldInherited("defaultDateTimeFormatInherited"));
+        }
+
+        checker().verifyEquals("Non-standard warning for DateTime (Date) not as expected.",
+                dtDateWarning, getNonStandardWarning("dateSelect"));
+
+        checker().verifyEquals("Non-standard warning for DateTime (Time) not as expected.",
+                dtTimeWarning, getNonStandardWarning("timeSelect"));
+
+        checker().screenShotIfNewError("Settings_Page_Error");
+    }
+
+    // Private helper to check the Site Validation report. Checks to see if the report should, or should not, contain
+    // the list of warnings.
+    private void validateSiteValidationReport(String scope, List<String> expectedWarnings, boolean shouldContain)
```

Comment:
Is there test that checks meta formats "date", "datetime" and "time" won't show up on the report?

Reply:
The NonStandardDateAndTimeFormatTest.testTextFormatValueOfDate uses the DATE, TIME, DATETIME and <none> values. I've added a call to check that the site validation report does not flag them.